### PR TITLE
Add a PR Template

### PR DESCRIPTION
**Pull request by adarsh**
Reason for Change
=================
* Encourages explanatory messages in pull requests

Changes
=======
* Add a `.gitmessage` file in RFC format

https://gist.github.com/lisawolderiksen/a7b99d94c92c6671181611be1641c733

### CODE REVIEW

**Comment by marcoroth** [comment]
I'm wondering if we should rather use a GitHub pull request template: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

**Comment by jonallured** [comment]
I'm a bit confused @adarsh - is this for PRs or commits? If it's for PRs then I agree GitHub PR templates seem like a better fit! If it's for commits then I'd wanna steer us away from anything that's triggered by git hooks. Not everybody uses git hooks (I don't!) and `--no-verify` is always available to bypass something like this. 🤷

**Comment by adarsh** [comment]
Whoops! Yeah I meant PR template. Updated in this PR.

File: pull_request_template.md
Comment (marcoroth) [inline pull_request_template.md:1]: I think usually the convention is to have this file at `.github/PULL_REQUEST_TEMPLATE.md`

Comment (adarsh) [reply pull_request_template.md:1]: Oh right. Fixed in 43e2e30

**Pull request merged**